bgpd/bgp_route.c

```diff
@@ -4438,8 +4438,12 @@ int bgp_nlri_parse_ip(struct peer *peer, struct attr *attr,
 					   BGP_ROUTE_NORMAL, NULL, NULL, 0,
 					   NULL);
 
-		/* Address family configuration mismatch or maximum-prefix count
-		   overflow. */
+		/* Do not send BGP notification twice when maximum-prefix count
+		 * overflow. */
+		if (CHECK_FLAG(peer->sflags, PEER_STATUS_PREFIX_OVERFLOW))
+			continue;
+
```

Comment:
Have you tested this and confirmed that a single notification is generated?

Reply:
of course :) I could even try to create a test case for this one.

Reply:
a test case would be awesome!

Reply:
@louberger added test case.